specification/network/resource-manager/Microsoft.Network/stable/2020-05-01/applicationGateway.json

```diff
@@ -2076,6 +2373,246 @@
     },
+    "ApplicationGatewayPrivateLinkConfiguration": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/ApplicationGatewayPrivateLinkConfigurationProperties",
+          "description": "Properties of the application gateway private link configuration."
+        },
+        "name": {
+          "type": "string",
+          "description": "Name of the private link configuration that is unique within an Application Gateway."
+        },
+        "etag": {
+          "readOnly": true,
+          "type": "string",
+          "description": "A unique read-only string that changes whenever the resource is updated."
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Type of the resource."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./network.json#/definitions/SubResource"
+        }
+      ],
+      "description": "Private Link Configuration on an application gateway."
+    },
+    "ApplicationGatewayPrivateLinkConfigurationProperties": {
+      "properties": {
+        "ipConfigurations": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ApplicationGatewayPrivateLinkIpConfiguration"
+          },
+          "description": "An array of application gateway private link ip configurations."
+        },
+        "provisioningState": {
+          "readOnly": true,
+          "$ref": "./network.json#/definitions/ProvisioningState",
+          "description": "The provisioning state of the application gateway private link configuration."
+        }
+      },
+      "description": "Properties of private link configuration on an application gateway."
+    },
+    "ApplicationGatewayPrivateLinkIpConfiguration": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/ApplicationGatewayPrivateLinkIpConfigurationProperties",
+          "description": "Properties of an application gateway private link ip configuration."
+        },
+        "name": {
+          "type": "string",
+          "description": "The name of application gateway private link ip configuration."
+        },
+        "etag": {
+          "readOnly": true,
+          "type": "string",
+          "description": "A unique read-only string that changes whenever the resource is updated."
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The resource type."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./network.json#/definitions/SubResource"
+        }
+      ],
+      "description": "The application gateway private link ip configuration."
+    },
+    "ApplicationGatewayPrivateLinkIpConfigurationProperties": {
+      "properties": {
+        "privateIPAddress": {
+          "type": "string",
```

Comment:
If user cannot input this value, please mark it as read only.

Reply:
subnet should be the only one field that user can choose. Is that correct?

Reply:
Nope. I was asked to change all this from last release and so I had to revamp .
We wanted to provide support of static Ip( see privateIPAllocationMethod) so ip address is configurable 

Resolving the comment as this is as expected